internal/manifest/os.go

```diff
@@ -465,12 +465,16 @@ func (p *OS) serialize() osbuild.Pipeline {
 	}
 
 	if p.Subscription != nil {
-		commands := []string{
-			fmt.Sprintf("/usr/sbin/subscription-manager register --org=%s --activationkey=%s --serverurl %s --baseurl %s", p.Subscription.Organization, p.Subscription.ActivationKey, p.Subscription.ServerUrl, p.Subscription.BaseUrl),
-		}
+		var commands []string
 		if p.Subscription.Insights {
+			// When using Insights, register using rhc instead of subscription-manager
+			commands = []string{fmt.Sprintf("/usr/bin/rhc connect -o=%s -a=%s --server %s", p.Subscription.Organization, p.Subscription.ActivationKey, p.Subscription.ServerUrl)}
```

Comment:
I think we might need to install rhc, as we don't install it by default. So if rhc would be specified maybe include it, wdyt @achilleas-k ? Otherwise include it by default but idk if that's ok.

Reply:
Probably so. It doesn't look like the code does anything to check for or add packages based on the Subscriptions being populated though. So we need to work that out so that it's not possible to end up with broken images (commands added but no supporting packages).

Reply:
We do add packages conditionally here.  The `getPackageSetChain()` function in this file does exactly that: checks what options are enabled and adds packages to the OS to support them:
https://github.com/osbuild/osbuild-composer/blob/356f261bcd9f3e36b9109e9c82b8c3c9d3875ac3/internal/manifest/os.go#L162-L211

Reply:
All that said, I'm not against `rhc` being installed unconditionally on RHEL.
In fact this kb article [0] implies it should be available on 8.4+.

[0] https://access.redhat.com/articles/6538061

Reply:
I went with an optional rhc bool, and adding required packages as needed.